kratos/tests/test_properties.py

```diff
@@ -71,10 +71,49 @@ def test_accessor(self):
         model_part= current_model.CreateModelPart("Main")
         properties = model_part.CreateNewProperties(1)
         self.assertFalse(properties.HasAccessor(KM.TEMPERATURE))
-        accessor_from_create = KM.Accessor.Create(properties, KM.TEMPERATURE)
+        accessor_from_set = properties.SetAccessor(KM.TEMPERATURE, KM.Accessor.Create())
         self.assertTrue(properties.HasAccessor(KM.TEMPERATURE))
         accessor_from_get = properties.GetAccessor(KM.TEMPERATURE)
-        self.assertEqual(id(accessor_from_create), id(accessor_from_get))
+
+        self.assertEqual(id(accessor_from_set), id(accessor_from_get))
+
+    def test_accessor_invalid(self):
+        current_model = KM.Model()
+        model_part= current_model.CreateModelPart("Main")
+
+        properties = model_part.CreateNewProperties(1)
+
+        with self.assertRaisesRegex(Exception, "Error: Trying to retrieve inexisting accessor for 'TEMPERATURE' in properties 1."):
+            accessor_from_get = properties.GetAccessor(KM.TEMPERATURE)
+
+    def test_accessor_consume_assigned(self):
+        current_model = KM.Model()
+        model_part= current_model.CreateModelPart("Main")
+
+        properties_1 = model_part.CreateNewProperties(1)
+        properties_2 = model_part.CreateNewProperties(2)
+
+        accessor_binding = KM.Accessor.Create()
+        
+        accessor_from_set = properties_1.SetAccessor(KM.TEMPERATURE, accessor_binding)
+        accessor_from_set = properties_2.SetAccessor(KM.TEMPERATURE, accessor_from_set)
+
+        with self.assertRaisesRegex(Exception, "Trying to get a consumed or invalid Accessor."):
+            accessor_from_get = properties_1.GetAccessor(KM.TEMPERATURE)
```

Comment:
Taking this into account (in this particular case the error would be thrown at the time the `GetAccessor` is called), I'd remove the return from the `SetAccessor` to always enforce the retrieval through the get.